cmd/scollector/collectors/cpu_windows.go

```diff
@@ -15,35 +15,66 @@ func init() {
-		AddTS(&md, "win.cpu.time_cstate", ts, v.PercentC2Time/1e5, opentsdb.TagSet{"cpu": v.Name, "type": "c2"}, metadata.Counter, metadata.Pct, "Percentage of time that the processor spends in the C-2 low-power idle state, which is a subset of the total processor idle time.")
-		AddTS(&md, "win.cpu.time_cstate", ts, v.PercentC3Time/1e5, opentsdb.TagSet{"cpu": v.Name, "type": "c3"}, metadata.Counter, metadata.Pct, "Percentage of time that the processor spends in the C3 low-power idle state, which is a subset of the total processor idle time.")
+		AddTS(&md, winCPU, ts, v.PercentPrivilegedTime/1e5, opentsdb.TagSet{"type": "privileged"}.Merge(tags), metadata.Counter, metadata.Pct, descWinCPUPrivileged)
+		AddTS(&md, winCPU, ts, v.PercentInterruptTime/1e5, opentsdb.TagSet{"type": "interrupt"}.Merge(tags), metadata.Counter, metadata.Pct, descWinCPUInterrupt)
+		AddTS(&md, winCPU, ts, v.PercentUserTime/1e5, opentsdb.TagSet{"type": "user"}.Merge(tags), metadata.Counter, metadata.Pct, descWinCPUUser)
+		AddTS(&md, winCPU, ts, v.PercentIdleTime/1e5, opentsdb.TagSet{"type": "idle"}.Merge(tags), metadata.Counter, metadata.Pct, descWinCPUIdle)
+		AddTS(&md, "win.cpu.interrupts", ts, v.InterruptsPersec/1e5, tags, metadata.Counter, metadata.Event, descWinCPUInterrupts)
+		Add(&md, "win.cpu.dpcs", v.DPCRate, tags, metadata.Counter, metadata.Event, descWinCPUDPC)
 	}
 	if num > 0 {
 		cpu := used / num
 		Add(&md, osCPU, cpu, nil, metadata.Counter, metadata.Pct, "")
 	}
+	if winCPUTotalPerfOS != nil {
+		v := winCPUTotalPerfOS
+		ts := TSys100NStoEpoch(v.Timestamp_Sys100NS)
+		AddTS(&md, winCPUTotal, ts, v.PercentPrivilegedTime/1e5, opentsdb.TagSet{"type": "privileged"}, metadata.Counter, metadata.Pct, descWinCPUPrivileged)
+		AddTS(&md, winCPUTotal, ts, v.PercentInterruptTime/1e5, opentsdb.TagSet{"type": "interrupt"}, metadata.Counter, metadata.Pct, descWinCPUInterrupt)
+		AddTS(&md, winCPUTotal, ts, v.PercentUserTime/1e5, opentsdb.TagSet{"type": "user"}, metadata.Counter, metadata.Pct, descWinCPUUser)
+		AddTS(&md, winCPUTotal, ts, v.PercentIdleTime/1e5, opentsdb.TagSet{"type": "idle"}, metadata.Counter, metadata.Pct, descWinCPUIdle)
+		AddTS(&md, "win.cpu_total.interrupts", ts, v.InterruptsPersec/1e5, nil, metadata.Counter, metadata.Event, descWinCPUInterrupts)
+		Add(&md, "win.cpu_total.dpcs", v.DPCRate, nil, metadata.Counter, metadata.Event, descWinCPUDPC)
+		AddTS(&md, winCPUCStates, ts, v.PercentC1Time/1e5, opentsdb.TagSet{"cpu": "total", "type": "c1"}, metadata.Counter, metadata.Pct, descWinCPUC1)
+		AddTS(&md, winCPUCStates, ts, v.PercentC2Time/1e5, opentsdb.TagSet{"cpu": "total", "type": "c2"}, metadata.Counter, metadata.Pct, descWinCPUC2)
+		AddTS(&md, winCPUCStates, ts, v.PercentC3Time/1e5, opentsdb.TagSet{"cpu": "total", "type": "c3"}, metadata.Counter, metadata.Pct, descWinCPUC3)
+	}
 	return md, nil
 }
 
+const (
+	winCPU               = "win.cpu"
```

Comment:
FYI, you can define consts in any block you want. Doesn't have to be globally like this. Can be in a function, an if block, etc.